Bayonetta Rework

jaccob101 opened the pull request:
Detronlegacy and I + discord feedback led to this. Nearly her entire moveset has been tweaked, a major point being adding more frame gaps in her cancels and more precise positioning. Gives her more options and more consistent ones but tones down some alpha stuff and makes her disadvantage far more vulnerable. Open to feedback and willing to bend on most changes. [Outdated changes:](https://docs.google.com/document/d/1T2qfDRQdxeL_X-TunFrLUyK2y9YozlhEIoo2xSFHirg/edit) 
[Requires Assets to Function:](https://github.com/HDR-Development/HewDraw-Remix/files/11360252/bayo.assets.zip) 
[bayo.assets.zip](https://github.com/HDR-Development/HewDraw-Remix/files/11360252/bayo.assets.zip)
[Pictures of changes frame by frame for witch twist + a few other hitboxes](https://drive.google.com/file/d/1jLRPNg6qichn4CyPQWRe8LjMufIk8UWI/view?usp=sharing)
**<<<=== BAYONETTA ===>>>**
General:
~ [+] Jump cancel window from 4 frames to 5 frames
~ [R] Special cancel count from 1 of each to 2 total
~ [R] ABK and DABK can utilize the jump cancels like WTW
~ [R] Visual effects and audio cue added on jump cancel success/failure.
~ [-] Jump cancel height gain nerfed
~ [+] ABK reuse window from 5 seconds to 10
~ [-] Bat Within Airdodges: Lag nerfed on all to 50 frames, bat animation active for 45 frames
~ [R] Aerial Bat Within now keeps Bayonetta stationary yet able to drift slightly in any direction
~ [R] Down taunts replaced with win animations, up and side taunts slowed to match Smash 4
~ [R] Gravity from 0.12 to 0.13 fall speed from 2.12 to 2.2, fast fall speed from 2.852 to 2.953. Ken gravity and incin fall speeds. 
~ [R] Recovery mechanic now always sets them to 1 if you get hit after whiffing, plus dAbk uses a resource now

Bullet Climax (Neutral Special):
~ [R] Shooting Angles: 6 -> 3 degrees (leg guns)
~ [+] Hitbox size: 0.7 -> 1.25 (uncharged) 
~ [-] Angle: 0 -> 361 (uncharged)
~ [+] Hitlag: 0.8x -> 0.4x (uncharged)
~ [-] KBG: uncharged -> 45 to 40, charged 210 -> 200
~ [+] FAF: 32 -> 26 (ending animation)
~ [+] Shield Cancel Frame: 46 -> 33
~ [R] Weak drifting and platform drops during charge implemented. Max speed .4, accel way lower
~ [+] Hold Frames: 80 -> 150
~ [-] Starting x-speed: 0.4x -> 0.3x

Witch Time (Down Special):
~ [+] Startup: 8 -> 5
~ [R] Counter Windows: 8-27 slowdown, 8-23 intangibility, 24-25 bat within -> 5-15 slowdown, 5-8 intangible, 16-21 bat within 
~ [R] Aerial Counter Windows: 8-27 slowdown, 8-23 intangibility, 24-25 bat within -> 5-15 slowdown, 5-21 bat within 
~ [+] FAF: 66 -> 50(g)/45(a)
~ [+] Activation Cancel Frames: WTi 18 -> 16 (sm4sh)
~ [-] Base Duration: 100 -> 75
~ [+] Staling Recovery Rate: 5% -> 8% (1.1.4 Sm4sh)
~ [+] Projectile Penalty: 40 -> 15
~ [+] Percent Scaling: 30% -> 50% (maxes out at ~330% as opposed to ~467%, evens out with beta at 125%)

Downward Afterburner Kick (Downward angled side special):
~ [+] SDI: 1.0x -> 0.8x
~ [+] Hitlag: 1.45x -> 1.1x
~ [+] Rotation: 0 -> 5
~ [+] BKB: 95 -> 105 (late)
~ [R] New Macro: Pressing the attack or grab button during abk startup functions as a new DABK input, first 5 frames and startup is standardized to 9 frames 
~ [R] Removed Macro: Tapping down during startup macro removed
~ [R] Tightened Macro: Quarter circle DABK window shortened from 20 to 12
~ [-] Now uses a recovery resource

Afterburner Kick (Upward angled side special):
~ [-] Gravity enabled during endlag, on whiff height will be lost and if no other resources remain you will reach max velocity far sooner
~ [+] SDI: 1.0x -> 0.8x
~ [+] Hitlag: 1.2x -> 1.1x
~ [-] Distance: 0.9x -> 0.81x distance 
~ [R] Angled distance: Horizontal distance now compensated for vertical distance, causing up angled to go less far, and down angled to go farther. 
~ [+] Hitbox size: 4 -> 4.5 units
~ [-] Startup 7 -> 8, angling easier
~ [R] Held hitbox: Instead of extending below her, now located inside her hips
~ [-] Ledge Grab Window: 20 -> 30
~ [-] FAF: Frame 33 -> 41?
~ [+] Cancel: Frame from 27 -> 26
~ [-] Bullet Arts Cancel: 27 -> 35
~ [+] Hitstun: -7/-5/-1 -> +2/0 (compensate for knockback)
~ [R] Effects: Now match trajectory
~ [R] Angles: No longer dynamic, 3 different angles like wolf flash. Angleable first 7 frames
~ [-] Conditional third ABK can't be cancelled out of on hit
~ [-] Damage: 7 -> 5/6/7/7.7
~ [R] Angles/knockback completely reworked. Move now functions as a dash combo extender/ callout, no longer scoops people. Move weakened up close, more consistent on tip
Early frames 8-10: 
~ [R] Angle: 50 -> 65 
~ [R] BKB: 85 -> 60
Early frames 11-13: 
~ [R] Angle: 69 -> 75 
~ [R] BKB: 70 -> 55
Mid frames 14-15: 
~ [R] Angle: 85 -> 35
Mid frames 16-18: 
~ [R] Angle 90 -> 35 
~ [R] KBG 112 -> 100 
~ [R] FKB 58 -> 70 
Final frames 19-20: 
~ [R] Angle: 90 -> 45
~ [R] KBG: 112 -> 100
~ [R] FKB: 58 -> 50

Heelslide (Grounded side special):
~ [+] On-Shield Speed Modifier: 0.5x -> 0.6x
~ [+] FAF: 66 -> 61
~ [R] Angle: 45 -> 42 (initial)
~ [R] BKB: 100 -> 101 (initial)
~ [R] Can now slide off ledge during frames 15->25. Holding B for BA will not slide off.
~ [R] Held hitbox now clears hitboxes allowing it to hit, combined with above makes move link more consistently pending rework
~ [R] Damage: 3.5 (held linking hitbox)
~ [-] BKB: from 60 to 64 (Kick)
~ [-] FAF 67 -> 72 (Kick)

Witch Twist (Up Special):
~ [R] Hitboxes: Move reworked. Much smaller, autolink angles to compensate. Move now effectively starts later in the animation.
~ [-] Height: 1.3x/1.2x -> 1.2x/1.1x/1.05x
~ [-] Speed multiplier: Vertical 0.5x -> 0.2x, horizontal 0.5x -> 0.4x
~ [+] FAF: Held BA from frame 45 of animation to frame 42 (3 frames faster)
~ [-] Ledge Grab: Animation frame 32 -> 42. Effect is negligible.
~ [+] Grounded Startup: 8 -> 7.  Has worse combo angle/kb/lag
~ [R] Aerial Scoop: Early hit is now a launcher, scoop hitbox does not come out for a few frames. Spacing and timing more necessary
~ [-] Cancel: Now only cancels if move is actually hit, 10 more frames of end lag on whiff
Grounded:
Scoop frames 7-9:
~Damage 3.0
~Angle 368
~Frame 9 arm hitbox clears
~SDI 0x
Rising frames 10-16: 
~Damage 3.0
~Angle 368
~Hitbox inversely disjointed 
~SDI 0x
Multihit frames 17-26: 
~Angle 367
~Damage 0.4 
~Rehit every 4 frames (2 to 3 times)
~Hitlag 0.5x
~SDI 0.3x
Finisher frames 27-28: 
~Damage 3.5 
~Angle 40/45 -> 50
~BKB 40-> 55
~KBG 40-> 55 
~Shrunk to no longer shark
~Hitlag 1.1x
~SDI 1.3x
Aerial:
Launcher frames 6-8: Damage 3.0
Angle 90/94 angle
~BKB 78
~KBG 62 
~Frame 8 the arm hitbox clears
Rising scoop frames 9-15: 
~Angle: 368
~Damage: 3 
Multihit frames 16-19: 
~Angle: 367
~Damage: 0.4
~Rehit: Every 4 frames (once)
~Hitlag: 0.5x
~SDI: 0.3x
Multihit frames 20-25: 
~Angle: 367
~Damage: 0.4
~Rehit: Every 3 frames (twice)
~Hitlag: 0.5x
~SDI: 0.3x
Finisher frames 26-27: 
~Angle: 40/45(2nd)
~Damage: 3.0
~Hitlag: 1.1x
~SDI: 1.3x

Neutral Attack:
~ [-] Jab no longer combos into rapid on whiff, worse shield pressure
~ [R] Jab 3 now a gentleman, Functions similarly to Corrin's jab, except in addition to A press timing the gentleman also uses the same special input as a jab like Zelda's.
~ [R] Can now hold to progress jab on-hit.
Neutral Attack 1:
~ [+] KBG: 6/8 -> 10
~ [R] BKB: 32/36 -> 31/41
~ [R] Angle: 50 -> 35/60/60
~ [-] +4 hitstun to +3 on tipper and +1 on middle hitbox, nerfed from +7/+9 on inner ones to +1 (double check) 
~ [+] Hitlag: 1.05x -> 1x
~ [+] SDI: 1.0x -> 0.8x
~ [R] Combo enabled 3 frames later from 9 to 12
~ [-] FAF from 19 -> 21
Neutral Attack 2:
~ [+] Hitlag: 1.25x -> 1.0x
~ [+] SDI: 1.0x -> 0.8x
~ [+] KBG: 8 -> 10
~ [R] BKB: 42/38 -> 46/43
~ [R] Angle: 50 -> 50/60
~ [-] +5 hitstun to +1 inner and +2 on tipper. Nerf makes all hits +5 compared to 7/9 before. (double check)
~ [R] Rapid jab hold Attack from frame 11 to 16, tap attack after frame 16 or press b any time to get jab 3 gentleman
Neutral Attack 3:
~ [+] FAF: 27 -> 24
~ [R] Angle: 40 -> 55
~ [+] Damage from 4 to 5.0
~ [R] BKB: 40 -> 71
~ [R] KBG: 20 -> 51
~ [+] Hitlag: 1.5x -> 1.2x
~ [R] No longer combos into rapid jab
~ [R] Bullet Arts re-enabled
Neutral Attack Loop:
~ [=] Damage: 0.2 -> .5
~ [R] Bullet arts disabled (may even out to less damage in addition to losing jab3)
~ [R] Angle: 361 -> 361/0
~ [R] BKB: 8/10 -> 5/15 
~ [R] KBG: 20 -> 15/25
~ [-] Size: 9 units long -> ~7.4
~ [-] Shield Setoff: 15x -> 10x
Neutral Attack Finisher:
~ [-] Size of Scoop: 18/16 long -> 15/13 units
~ [+] Scoop box positioned closer to bayo and hits 1 unit shorter vertically.
~ [+] Finisher: Pulled closer to bayo and combined into one hitbox
~ [-] Size: 10 long -> ~5.3 units
~ [+] Hitlag: 1.5x -> 1.2x 
~ [+] Sdi: 0.7x -> 1.0x 
~ [-] Angle: 40 -> 45
~ [-] FAF: 41 -> 48

Side Tilt 1:
~ [+] Hitlag: 1.5x -> 1.1x
~ [+] SDI: 1x -> 0.5x. 
~ [-] Hitstun: +3 -> 0
~ [R] Hold stick up or press b to get the upwards angled kick, else continues into horizontal kick. No longer a 3-hit combo
Side Tilt 2 (horizontal): 
~ [+] Damage from 3 to 5
~ [-] Middle hitbox removed
~ [R] Angles: 48/62 -> 66 
~ [R] KBG: 15/25 -> 50
~ [R] BKB: 42/46 -> 81
~ [+] Hitlag: 1.5x -> 1.2x 
~ [+] SDI: 1.1x -> 0.8x
~ [+] FAF: 43 -> 32
~ [R] Doesn't enable combo into ftilt 3.
Side Tilt 3 (vertical):
~ [=] BKB: 70 -> 75
~ [+] Hitlag: 1.4x -> 1.2x
~ [+] SDI: 1.1x -> 0.8x
~ [=] Hitstun: +2 -> 0
~ [-] Removed hit cancels

Down Tilt:
~ [+] SDI: 1.1x -> 1x 

Aerials:
~ [-] Cancels: From instantly on-hit to delayed until hitbox clear / sour hitbox comes out. 
Neutral Aerial:
~ [R] Now uses tipper system, hitboxes shrunk to match visual better. Sends at more benificial angle even though the knockback is weaker, better for edgeguarding but hopefully not for easy cheese kills off abk strings
~ [-] Damage: 9 -> 9/8.5/8 
~ [+] Angle: 55 -> 55/50
~ [-] BKB: 55 -> 45
~ [+] Hitlag: 1.4x -> 1.0x
~ [+] SDI: 1.1x -> 1x
~ [-] Sourspot: Starts earlier at frame 12 after foot goes behind her
~ [-] Cancel: Frame 6+ -> 12+ 
Held Neutral Air:
~ [R] Rehit: 0 -> 9 
~ [+] Loop: 4x -> 8x
~ [+] KBG: 60 -> 65
~ [+] BKB: 25 -> 30
~ [=] SDI and Hitlag: 1.25x/0.75x -> 1.0x
Back Aerial:
~ [-] Startup: 8 -> 9
~ [+] Middle now highest priority, sweet spot second priority
~ [+] Tip hitboxes no longer reverse hit
~ [+] Damage: 9.0 to 10.0 (middle), 8.5/9 -> 8 (weak sourspots)
~ [+] BKB: 15 -> 20(sourspots)
~ [-] KBG: 136/128 -> 120
~ [-] Size: 3.5 -> 3.0 (sour), 3.5 -> 1.5 (weak leg sour). Sweetspots skinnier, still has the same horizontal range
~ [-] Angle: 40 -> 50 (weak sourspots)
~ [-] Cancel: Frame 8+ -> 14+
Up Aerial:
~ [+] Hitlag: 1.4x -> 1.2x/1x
~ [+] SDI: 1.1x -> 1x
~ [R] Sweetspot from 8 - 10, sour from 12 to 16
~ [+] Cancels into BA 1 frame earlier
~ [-] FAF: 24 -> 27
~ [-] Autocancel: 22 -> 25
~ [-] Cancel: 8+ -> 12+
~ [R] Angle 71 -> 77, 55
~ [+] KBG of sweetspot from 82 -> 105 (sweetspot)
~ [-] Damage: 8 -> 7 (sour)
~ [-] BKB: 45 -> 50 (sour)
~ [R] Hitbox size now a cone, no longer offset from the leg
Held:
~ [+] Hitbox now on both feet
~ [+] Loop: 4 -> 8x
~ [-] Size: 3.8 -> 3.5
~ [-] Hitlag: 0.75x -> 1.0x
~ [+] SDI: 1.25x -> 1.5x
~ [R] Rehit: 0 -> 9

Forward Aerial:
~ [R] Y movement multiplier from 1.4 to 1.3, horizontal movespeed decreased on hit
~ [-] No longer checks if fair 1 hit
~ [R] Changes overall incentivize using the full move instead of repeatedly cancelling, to make up for improved frame data
Forward Aerial 1:
~ [-] Hitbox: Knee hitbox skinnier, no longer air only
~ [+] Startup: 6 -> 5
~ [-] Activity: 6-8 -> 5-6
~ [+] FAF: 30 -> 28
~ [+] Hitlag: 1.2x -> 1x
~ [+] SDI: 0.9x -> 0.8x
~ [-] Hitstun: +2 -> -4 
~ [R] Angles: 55/40/75/50/65 -> 60/45/75/55/65 (compensates for weaker hitstun and side-carry potential to make it still link)
Forward Aerial 2:
~ [-] Damage: 3.5 -> 2
~ [+] Startup: 7 -> 5
~ [-] Activity: 7-9 -> 5-6
~ [+] FAF: 37 -> 35
~ [+] Hitlag: 1.2x -> 1x
~ [+] SDI: 1.1x -> 1.0x
~ [-] Hitstun: +2 -> -4 
~ [-] No longer cancellable into aerials
~ [-] Cancel: 7+ -> 11+
Forward Aerial 3:
~ [+] Damage:  4.5 -> 6
~ [R] Angle: 60 -> 50 (Better for edgeguarding, but hurts late percent combos in combination with increased damage)
~ [+] Startup: 12 -> 7
~ [+] FAF: 45 -> 40
~ [+] Hitlag: 1.7x -> 1.2x
~ [+] SDI: 1.1x -> 1x
~ [-] Cancel: 12+ -> 11+ (4 after hitbox comes out)

Down Aerial:
~ [+] Sweetspot out 1 more frame
~ [+] Fall Speed: -4.2 -> -3.7
~ [+] Angle: 270 -> 230 (Ground-only tipper, compensates for landing hit not going as far down her leg)
~ [R] BKB: 90 -> 85 (Ground-only, removes ground bounce preventing people from bouncing away from the second hit)
~ [R] Landing hitbox added back. 
~ [R] 0 to 3 active frames 
~ [-] BKB from 80 to 74
~ [-] Cancel: 18+ -> 26+

Up Smash:
~ [-] Hitboxes shrunk to cover less space
~ [+] Angle: 86 -> 89
~ [+] BKB: 32 -> 38/36/35/34/32 (weaker the higher up it hits)

Down Smash:
~ [-] Hitstun: 5 -> 4

Forward Smash:
~ [R] Size on fist reduced to match vanilla, late hit added after 2 frames that has a smaller hitbox and weaker damage
~ [R] Fixes angled smashes not being changed, only unangled was hooked
~ [-] Angle: 361 -> 49 (angled)
~ [-] KBG: 105 -> 100
~ [-] Damage: 16 -> 15 (late hits)

**Forward Throw:**
~ [+] Angle from 49 to 45. 
~ [-] KBG from 150 to 140
~ [+] Damage shifted, first hitbox from 7 to 6% and finisher from 3 to 4%. This makes it kill inkling at ~183% at ledge on fd, a bit worse than vanilla as compared to 230%+ in beta.


Up Special Distance Comparison (Jump Cancelled):
Notes: Basically a slight distance buff at worst scenario due to having 2 instead of 1, and a big nerf in her best-case scenarios due to each move losing effectiveness and getting nerfed speed/framedata. Bayonetta will still have a good recovery range but if in disadvantage the opponent can easily move to edgeguard her.
Before:
![prenerf wtw](https://user-images.githubusercontent.com/122749442/234949054-baa3b04c-debe-4d0c-aefd-d91c9e54a12d.png)
After:
![post nerf wtw](https://user-images.githubusercontent.com/122749442/234949081-b94f2363-d837-4d40-a9e6-9a63f1fb0318.png)
Both Up Specials + jumps:
Before:
![double wtw distance prenerf](https://user-images.githubusercontent.com/122749442/234950811-b89a763f-69f6-45f4-a052-c6eb93c7169f.png)
After:
![double wtw distance postnerf](https://user-images.githubusercontent.com/122749442/234950828-b184e311-bc47-4530-aa7c-7ed5bf9a8157.png)
HITBOXES:
Neutral Special:
Before:
![2023021002124200-0E7DF678130F4F0FA2C88AE72B47AFDF](https://user-images.githubusercontent.com/122749442/218044315-0bf1d120-980f-461f-90ca-0a7750e22c33.jpg)
![2023021002125000-0E7DF678130F4F0FA2C88AE72B47AFDF](https://user-images.githubusercontent.com/122749442/218044321-7529ad6c-1d38-4258-9f63-6fdde33fefb4.jpg)
After:
![2023021000010200-0E7DF678130F4F0FA2C88AE72B47AFDF](https://user-images.githubusercontent.com/122749442/218044453-6833e8f8-735c-437e-a281-d80583dc9087.jpg)
![2023021000011200-0E7DF678130F4F0FA2C88AE72B47AFDF](https://user-images.githubusercontent.com/122749442/218044466-291aeaa3-f5ec-40aa-b62d-c8a6c4bc9107.jpg)
Side Special:
Aerial:
Before:
![abknightly](https://user-images.githubusercontent.com/122749442/223513166-556b0b74-2069-4519-876a-6c70e4267866.png)
![image](https://user-images.githubusercontent.com/122749442/218040045-8287003b-a75d-44df-b843-503160c2d36f.png)
After:
![abkpr](https://user-images.githubusercontent.com/122749442/223513173-53f9d9da-1486-4d82-b041-859a0f6759ed.png)
![image](https://user-images.githubusercontent.com/122749442/218010264-5a94ce5b-be9e-4340-8cfe-9e6d17c39cbf.png)
Up Special:
ground:
Before:
![image](https://github.com/HDR-Development/HewDraw-Remix/assets/122749442/6849f2a7-a714-4bcb-816d-913787e1cb1b)
![2023021002112800-0E7DF678130F4F0FA2C88AE72B47AFDF](https://user-images.githubusercontent.com/122749442/229404618-a60b9882-ecca-4509-ba5e-04e650159b00.jpg)
After:
![wtw1pr](https://user-images.githubusercontent.com/122749442/229404684-5ff88fd0-ce50-4ae4-9e16-d14288530373.png)
![wtw3pr](https://user-images.githubusercontent.com/122749442/229404728-5d81e174-aab5-41ed-9a95-79c0eec2d7fe.png)
Air:
Before:
![image](https://github.com/HDR-Development/HewDraw-Remix/assets/122749442/ade7d754-1b2c-4252-bf5e-4b3196b9bf69)
After: (before matches ground)
![airwtw1pr](https://user-images.githubusercontent.com/122749442/229404708-5a83f630-f300-43c4-bed1-251f9977ae70.png)
![airwtw3pr](https://user-images.githubusercontent.com/122749442/229404717-59b4a7e3-8bf8-4b36-9307-4a1b05dde2cc.png)
Both:
Before: 
![image](https://github.com/HDR-Development/HewDraw-Remix/assets/122749442/63b80594-09f6-4743-96ce-294662126fe2)
![image](https://github.com/HDR-Development/HewDraw-Remix/assets/122749442/5c1e2a63-09e2-45fc-b04b-7cf86086e347)
![image](https://github.com/HDR-Development/HewDraw-Remix/assets/122749442/982244cb-00f9-4bf1-a47f-876ea0708b1f)

After:
![revamp wtw rise](https://github.com/HDR-Development/HewDraw-Remix/assets/122749442/dc2e7d5a-25f1-4dd2-8b30-2dee34de540a)
![revamp wtw multihit](https://github.com/HDR-Development/HewDraw-Remix/assets/122749442/b85b4334-b154-4258-9bcc-3a060e77bc2c)
![revamp wtw finisher](https://github.com/HDR-Development/HewDraw-Remix/assets/122749442/939a4805-382f-4e15-a2df-b3b95b4ac72e)

Rapid Jab:
Before:
![image](https://user-images.githubusercontent.com/122749442/218039035-49da1472-6f4b-47de-aa6d-61d8f82e2745.png)
![image](https://user-images.githubusercontent.com/122749442/218038978-300b94b7-52ae-4f56-a8c2-61270cfb8117.png)
![image](https://user-images.githubusercontent.com/122749442/218039172-22383d58-aecf-4c7e-b295-832bae0a7427.png)
After:
![image](https://user-images.githubusercontent.com/122749442/218025738-45845b11-a8a1-4084-a883-2cf9ac31432a.png)
![rapidjabpr](https://user-images.githubusercontent.com/122749442/230695144-98804d69-87eb-4dbb-8908-6f2683db3b31.png)
![image](https://user-images.githubusercontent.com/122749442/218025792-845b75e4-6a51-4dfd-b82d-af2c2018d31d.png)
FTilt 2:
Before: 
![ftilt2nightly](https://user-images.githubusercontent.com/122749442/223513980-c3cec842-146b-414b-8802-f47d68e46bd2.png)
After:
![ftilt2pr](https://user-images.githubusercontent.com/122749442/223513985-3e6ded28-04ca-4703-a0be-c5b8fe6555ce.png)
Nair:
Before:
![nairnightly](https://user-images.githubusercontent.com/122749442/223514389-a9f32c6e-5296-4207-83a9-a3a28c88cb5b.png)
![latenairnightly](https://user-images.githubusercontent.com/122749442/223514301-eb074704-0d07-44b1-9715-4c57e033f8d0.png)
After:
![nairpr](https://user-images.githubusercontent.com/122749442/223514286-59262833-b3e1-4ca2-adcd-568025c71692.png)
![latenairpr](https://user-images.githubusercontent.com/122749442/223514308-f8749ea5-e4ac-4b4d-b182-1cee0d1991eb.png)
Bair:
Before:
![bairnightly](https://user-images.githubusercontent.com/122749442/223514883-93fd4415-d102-4f39-a8fd-ac7ca70b1bdf.png)
After:
![bairpr](https://user-images.githubusercontent.com/122749442/232836894-c24202fb-35db-4893-b4e3-bacfeac02463.png)
UAIR:
Before:
![uairnightly](https://user-images.githubusercontent.com/122749442/223515242-c996c756-44be-477c-94d7-86532177929f.png)
![lateuairnightly](https://user-images.githubusercontent.com/122749442/223515262-488a7acc-cc5d-4881-b051-3bcce29c4dad.png)
![2023021002140600-0E7DF678130F4F0FA2C88AE72B47AFDF](https://user-images.githubusercontent.com/122749442/218044144-977f047a-e4ee-4491-9836-c585029fa108.jpg)
After:
![uairpr](https://user-images.githubusercontent.com/122749442/235376075-93cdcafa-e541-47bc-b0d7-ff712f8f3aef.png)
![uairlate](https://user-images.githubusercontent.com/122749442/235376080-c5eac71a-452c-4255-9099-5fb5f8f15512.png)
![uairend](https://user-images.githubusercontent.com/122749442/235376083-d96aef0a-e5c5-4e59-b38d-98ee525bfd1b.png)
![image](https://user-images.githubusercontent.com/122749442/218034792-0e73aa9e-77cc-4c63-a6be-0d2ed7fd04b3.png)
Fair:
Before:
![image](https://user-images.githubusercontent.com/122749442/218038182-acd6359d-5085-4321-b603-ce5a8dcbc7f7.png)
After:
![revamp fair 1](https://github.com/HDR-Development/HewDraw-Remix/assets/122749442/284dad9a-f725-4b3d-b2c7-7290fa97a16a)
Dair:
After: (none before)
![image](https://user-images.githubusercontent.com/122749442/218034905-e369e8be-14eb-403b-8b43-14e24ac0ba1a.png)
Up Smash:
Before:
![hi42beta](https://user-images.githubusercontent.com/122749442/225088783-e3519b6d-4801-47e1-a731-8a42c8119cb2.png)
![hi4endbeta](https://user-images.githubusercontent.com/122749442/225088816-bb1c3129-df01-4119-a31d-25b1cf3de2d5.png)
After:
![hi42pr](https://user-images.githubusercontent.com/122749442/225088796-f7e6e87e-f93d-4cde-bb46-b0b284c260b8.png)
![hi4endpr](https://user-images.githubusercontent.com/122749442/225088827-78f3763d-b887-455b-877e-aab5f684d865.png)
Forward Smash:
Unangled:
Before:
![s4sbeta](https://user-images.githubusercontent.com/122749442/225089805-73f86ff0-f8ed-4b31-ab7a-db925bb5cb42.png)
After:
![s4spr1](https://user-images.githubusercontent.com/122749442/225089822-4d0e7894-1e48-44d2-9334-9cdbf88eb615.png)
![s4spr2sour](https://user-images.githubusercontent.com/122749442/225089838-6c29ed04-dd28-4629-b1d7-9ed178faf76a.png)
High:
Before:
![s4hibeta1](https://user-images.githubusercontent.com/122749442/225089917-0d864216-8da4-43cc-98b5-02
![s4hibeta2](https://user-images.githubusercontent.com/122749442/225089933-072c6267-cf22-4449-a768-1caf0149c9b0.png)
5b8aade6d1.png)
After:
![s4hipr_interpolation hitbox](https://user-images.githubusercontent.com/122749442/225089955-26252c53-bfbe-4754-b07d-4d9157b1eec9.png)
![s4hipr1](https://user-images.githubusercontent.com/122749442/225089972-45371461-56c1-4251-9ba2-3ebd03f0bdc0.png)
![s4hipr2](https://user-images.githubusercontent.com/122749442/225089981-145f9f88-16bb-41be-8804-d3576c74c1a9.png)
Low:
Before:
![s4lwbeta1](https://user-images.githubusercontent.com/122749442/225090014-fcac1eae-8bfe-4723-b44b-9637d2c187e3.png)
![s4lwbeta2](https://user-images.githubusercontent.com/122749442/225090040-13257f10-1327-48f9-8d61-c30872c5b49b.png)
After:
![s4lwpr_interpolation](https://user-images.githubusercontent.com/122749442/225090068-b2c99d78-3b29-47ef-896a-bd2a2ebd4355.png)
![s4lwpr1](https://user-images.githubusercontent.com/122749442/225090095-4c17ff9a-1df8-4687-aa18-4e2b9bfb9f3c.png)
![s4lwpr2](https://user-images.githubusercontent.com/122749442/225090113-8cb21c3c-7fb4-4f4d-96d4-8caf7ad0e19a.png)

Disclaimer: Many edits could become more concise if status scripts were utilized and someone more experienced made it more efficient. Bugs should be ironed out hopefully as of freeze. Effects should be fixed too.

jaccob101 committed:
Detronlegacy and I have collabed on Bayonetta changes with feedback from other players in the Bayonetta channel. Changelist to be updated: https://docs.google.com/document/d/1T2qfDRQdxeL_X-TunFrLUyK2y9YozlhEIoo2xSFHirg/edit

hdr-balance-backroom commented:
Witch Time:
- Revert animation changes, use motion rating for the relevant scripts instead
- Increase animation duration from what it currently is, as the move is very spammable

jaccob101 commented:
I tried using fsms to change the frame data of witch time but it broke the baked movement and made the move not function. I changed the FAF from 33 to 39, so 24 frames of lag after the witch time window ends.

jaccob101 commented:
If you have a suggestion on how to use FSMs without having to change the animation in the first place to remove the movement then I can try it

WuBoytH commented on romfs/source/fighter/bayonetta/param/vl.prcxml:
```diff
@@ -17,6 +17,11 @@
   </struct>
+  <list hash="param_specialn_bullet">
+	  <struct index="0">
+		  <int hash="0x0d76ae442e">3</int>
```
I would suggest not using `tab` to space out lines in the prcxml, use spaces instead

jaccob101 commented:
After trying it disabling move true on the witch time just makes it teleport back after moving during the animation so I would need more guidance if the aenimation needs to be the vanilla one

hdr-balance-backroom commented:
This will be reviewed further after the next beta release

hdr-balance-backroom commented:
Please put a code freeze on this (no more commits) until we can properly evaluate

jaccob101 commented:
Ok i'll abstain but i have a commit ready if you want me to remove the jabs and ftilt 3 extra hitstun and stuff

hdr-balance-backroom reviewed (changes requested):
- Remove .vs files from PR
- Revert jumpsquat to 5 frames
  - this change was made to offset the potency of characters with especially strong aerial tools
- Revert model scale increase
- Lower knockback on jab2 to reduce frame advantage (currently +13/+9 at 0% for the inner/outer hitboxes)
- Remove additional hitstun on ftilt1 if it is not absolutely necessary to combo into ftilt2/3
- Revert utilt and dtilt startup decreases
  - Bayonetta should have below average neutral tools to offset her punish game
- Change nair hitlag to 1.0 on all hits
- Remove held nair additional hitstun and change hitlag/SDI to 1.0
- Revert fair landing lag to 9 frames
- Revert bair landing lag to 10 frames and change the front leg and hip hitboxes to reverse hit
- Remove additional hitstun on held uair
- Remove dair shieldstun multiplier and change hitboxes to reverse hit
- Change Heel Slide to send into fall instead of tumble when sliding off an edge (may have to edit the status script)
- Reduce bullet art ABK hitbox size, move to only cover lower leg/hip area
- Revert DABK hitbox size
  - this move should be intentionally negatively disjointed
- Potentially remove the directional input for dabk in favor of the new attack+special input
- Standardize Witch Time startup to frame 5 for both ground and air
- Reduce fthrow voice line chance to around 10-20%

jaccob101 commented:
I've been playing more with the requested changes and I just would like to ask if the uptilt startup and the jump squat are non-negotiable? I nerfed up tilt's lingering hitboxes by 33% and previously compensated the FAF so that it wouldn't be spammable in neutral. If not 8 frames of startup 9 frames would be enough to improve the feel. Also, the majority of her aerial options are nerfed and most of her rising aerials don't hit grounded opponents. None of her aerial options are safe on shield and now she gets punished harder for mistakes due to her higher gravity/fall speed. I believe the start-up change would not improve her neutral much, just making it feel better to move with her and wavedash. She already will generally use witch twist out of shield because it's the only one that can consistently hit low opponents while also being far faster than her next fastest oos (10 frames). Also I agree with the rest of your feedback as many of those changes were unnecessary and should have been reverted after testing. The landing lag changes were an attempt to make her moves safer on shield as back air is far more committal now for less reward, especially after the jump revert. The fair change did improve the combo potential of landing fair 3 so I understand that 2 frames off might be more impactful than intended. I don't want to remove her other dabk inputs so it doesn't alienate bayo mains that have muscle memory. Thank you for looking at this pr and apologies for not realizing I should have not put an unfinished pr on here, just wanted people to be able to download it from the launcher as draft prs do not show up.

hdr-balance-backroom commented:
> If not 8 frames of startup 9 frames would be enough to improve the feel. None of her aerial options are safe on shield and now she gets punished harder for mistakes due to her higher gravity/fall speed. I believe the start-up change would not improve her neutral much, just making it feel better to move with her and wavedash

Again, the point is that her neutral tools should be intentionally worse than average because her punish game is well above average. Jumpsquat also plays into this, she can get full combos off stray hits easier than other characters with jump/aerial/special cancels. This rework already changes so much, the current implementation need to sit a while before making any more changes. If utilt ends up being weaker than intended then it can be reevaluated at a later point

jaccob101 commented:
Ok I fixed every bug I could find, cleaned up effects to match shortened distances and smaller hitboxes, and normalized the jab hitstun values. It should have a lot of the bs ironed out so I feel it's ready for a retest

jaccob101 committed:
Restore

Messed up something on github desktop, tried to fix by undoing last commit but it undid all of them?

jaccob101 commented:
> Witch Time:
> 
>     * Revert animation changes, use motion rating for the relevant scripts instead
> 
>     * Increase animation duration from what it currently is, as the move is very spammable

---
Late but figured out I needed to set the FSM as a float/float instead of an int/int to make this work. Animations have been removed from the folder, but it still needs a motion list edit.

MadVillainLordQ commented on fighters/bayonetta/src/acmd/aerials.rs:
```diff
@@ -222,49 +213,45 @@ unsafe fn bayonetta_attack_air_f3_game(fighter: &mut L2CAgentBase) {
-        ATTACK(fighter, 3, 0, Hash40::new("kneer"), 8.5, 40, 70, 0, 40, 3.5, 2.0, 0.0, 0.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_POS, false, 0, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_M, *COLLISION_SOUND_ATTR_KICK, ...
+        ATTACK(fighter, 2, 0, Hash40::new("legl"), 9.0, 40, 131, 0, 15, 3.4, 0.0, 0.0, 0.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_POS, false, 0, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_M, *COLLISION_SOUND_ATTR_KICK, ...
+        ATTACK(fighter, 0, 0, Hash40::new("kneel"), 10.0, 40, 131, 0, 15, 4.2, 0.1, 0.0, 0.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_POS, false, 0, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_M, *COLLISION_SOUND_ATTR_KIC ...
+        ATTACK(fighter, 1, 0, Hash40::new("kneel"), 12.0, 40, 121, 0, 15, 4.7, 6.7, 0.0, 0.0, None, None, None, 1.2, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_POS, false, 0, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_L, *COLLISION_SOUND_ATTR_KIC ...
```
Hitlag Modifier should be 1.0

hdr-balance-backroom commented:
Requesting a full code freeze. Please do not make any more commits, including merging in dev

jaccob101 commented:
just saw will do

Updated to latest beta just realized that counts as merging sorry
-----------
Cant figure out how to make it work again and my backup was out of date, so this is my latest working copy. Feel free to tell me what to cut out of it im tired at this point. Can read commit name and description for more details, sorry for unintentionally messing it all up hoping i didnt disturb anything

jaccob101 committed:
Update / Restore 

properly this time, back to code that was right before freeze

jaccob101 committed:
Backup was outdated, cant figure out how to bring back old commit. Apologies for ineptitude. This was my only backup not months outdated, might have forgotten to remove some things. Features bug fixes, logic improvements, less intrusive effects, slight tweaks, updated formatting to my limited knowledge. If something extraneous is in here please tell me to cut it out, im too tired for this.

My fault for updating it wrongly somehow. I use this on my personal setup otherwise I would not bother people by keeping it up to date. No idea if this was still in consideration but if it was please dont let this break that chance, send me a message and tell me what to do. Removed some stupid errors i noticed immediately

hdr-balance-backroom reviewed (dismissed):
- Reduce Bayonetta's ability to gain back two resources when sent into tumble to just one resource
- Increase the SDI multiplier on held uair hitboxes from 1.0x -> 1.5x
- Decrease on hit frame advantage of held nair hitboxes (currently sits at +8/+14 at 0/100). Suggested amount is from +5-+6 at 0% and +11-+12 at 100%
- Upwards ABK gives Bayonetta less gravity on hit than on whiff
- Remove ABK's ability to cancel 5 frames sooner when hitting a shield

jaccob101 committed:
BR Feedback: BA Nair hitbox bkb down 5 to lower base frame advantage from 8 to 6 on-hit, BA Uair sdi from 1x to 1.5x, now getting hit after burning 1 resource will always leave bayo with 1 special, hit flag now doesnt check for shield infliction, bug fixed where abk doesnt cancel on hit making uabk lose more height on hit

jaccob101 commented:
> 

Done I believe. 
Held N-Air: BKB from 35 to 30 (should lower adv by 2 frames)
Held U-Air: Sdi from 1x to 1.5x
Resources recovered standardized, hit flag no longer checks for shield-hits
I believe the U-Abk losing height on hit was the result of a bug where I improperly checked for the hit-flag, making it so it would transition into fall kinetic kind where normally the move would have been cancelled already (unless you meant the opposite)

LMK if you have more issues w it

SkewedAskew reviewed (approved):
'Twas a long journey, good job
